easybuild/tools/filetools.py

```diff
@@ -763,6 +763,18 @@ def apply_patch(patch_file, dest, fn=None, copy=False, level=None):
     apatch = os.path.abspath(patch_file)
     adest = os.path.abspath(dest)
 
+    if apatch.endswith(".gz"):
```

Comment:
The extract_cmd dict might be helpful. Maybe we should store it somewhere accessible and use the keys?